ddmrp/models/stock_buffer.py

```diff
@@ -702,11 +702,19 @@ def _compute_dlt(self):
             elif rec.buffer_profile_id.item_type == "distributed":
                 rec.dlt = rec.lead_days
             else:
-                rec.dlt = (
-                    rec.product_id.seller_ids
-                    and rec.product_id.seller_ids[0].delay
-                    or rec.lead_days
+                sellers = rec.product_id.seller_ids
+                sellers_variant = sellers.filtered(
+                    lambda s: s.product_id == rec.product_id
                 )
+                sellers_no_variant = sellers.filtered(lambda s: not s.product_id)
+                if sellers_variant:
+                    rec.dlt = sellers_variant[0].delay
```

Comment:
If there is multiple suppliers, should they be sorted by delay so it is always the lowest or highest that is selected ?
Same question for the assignations of `rec.dlt` below.

Reply:
That's a good question and if you strictly stick to the theory that's probably true but in reality 99% of the cases the first seller will be used, because in Odoo the first matched seller is used for procurement (thus for the buffer as well) and they are [sorted by sequence by default](https://github.com/odoo/odoo/blob/13.0/addons/product/models/product.py#L711). So, we take the most common use case as the default here.

Reply:
@LoisRForgeFlow Thank you for your answer.